sdk/python/tests/compiler/testdata/long_pipeline_name.yaml

```diff
@@ -16,10 +16,9 @@ apiVersion: tekton.dev/v1beta1
 kind: PipelineRun
 metadata:
   annotations:
-    pipelines.kubeflow.org/pipeline_spec: '{"inputs": [{"default": "[\"a\", \"b\",
-      \"c\"]", "name": "arr", "optional": true, "type": "typing.List[str]"}], "name":
-      "Some very long name with lots of words in it. It should be over 63 chars long
-      in order to observe the problem."}'
+    pipelines.kubeflow.org/pipeline_spec: '{"inputs": [{"name": "arr", "optional":
```

Comment:
You should not change this line. If you run `make unit_test GENERATE_GOLDEN_YAML=True` again this issue should be fixed

Reply:
Strange everytime I run `make unit_test GENERATE_GOLDEN_YAML=True` I got the same result, maybe python or sdk version issue...

Reply:
I saw from travis it works for python 3.6 but not for 3.7 and 3.8.

Reply:
This test is only working on 3.7 and 3.8, so we skipped this on python 3.6. Are you using Python 3.7+ to run `make unit_test GENERATE_GOLDEN_YAML=True`?

Reply:
if you revert the changes in this file, then it should able to pass the test.

Reply:
I run `make unit_test GENERATE_GOLDEN_YAML=True` and the issue still exist. Anyway, rollback it to fix the travis issue for now but everytime `make unit_test GENERATE_GOLDEN_YAML=True` will generate a diff for me.

Reply:
Yes I am using `Python 3.9.0`